app/controllers/api/databases.php

```diff
@@ -2564,22 +2868,88 @@ function updateAttribute(
 
         $filterQueries = Query::groupByType($queries)['filters'];
 
-        if ($documentSecurity && !$valid) {
-            $documents = $dbForProject->find('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $queries);
-            $total = $dbForProject->count('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $filterQueries, APP_LIMIT_COUNT);
+        // TODO: Remove this when we have a better way to handle nested attribute queries
+        foreach ($filterQueries as $key => $query) {
+            if (\str_contains($query->getAttribute(), '.')) {
+                unset($filterQueries[$key]);
+            }
+        }
+
+        $documents = Authorization::skip(fn () => $dbForProject->find('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $queries));
+
+        $documentSecurity = $collection->getAttribute('documentSecurity', false);
+        $validator = new Authorization(Database::PERMISSION_READ);
+        if (!$validator->isValid($collection->getRead())) {
+            $total = $documentSecurity
```

Comment:
It looks like there's still a possibility to get a incorrect `$total` if filtering on nested data. A query like

```
equal("user.name", "Steven")
```

resulted in a greater `$total` than what was in `$documents`. Maybe because of the `$processDocument` post processing?

![image](https://user-images.githubusercontent.com/1477010/228994769-a31016d8-0e77-47d5-bb03-90c5067da615.png)

Reply:
What was the permissions configuration to get this error? The nested queries would have been ignored, so the issue must be elsewhere

Reply:
@abnegate, the Users collection has document-level permissions. Regardless, though, this was done in admin mode.

Reply:
@stnguyen90 This should be resolved now

Reply:
@abnegate what was resolved and how? I'm still seeing the same behavior 🧐 

![image](https://user-images.githubusercontent.com/1477010/230965038-fbe27845-11f9-4d18-92fe-18a2989d83ed.png)

Reply:
@stnguyen90 Looks like nested queries weren't being filtered out properly. I've updated so that queries on nested attributes fail during validation, until it's supported. It seems like better behavior than just ignoring the query